sql/persisted_variable.cc

```diff
@@ -332,17 +334,19 @@ bool Persisted_variables_cache::set_variable(THD *thd, set_var *setvar) {
     uint dummy_err;
     String bool_str;
     if (setvar->value) {
-      res = setvar->value->val_str(&str);
+      setvar->var->persist_only_update(thd, setvar, str.ptr());
+      res = &str;
+
       if (system_var->get_var_type() == GET_BOOL) {
-        if (res == nullptr ||
-            check_boolean_value(res->c_ptr_quick(), bool_str)) {
+        if (check_boolean_value(res->c_ptr_quick(), bool_str)) {
```

Comment:
Not sure about this. The Sys_var_bool::saved_value_to_string() is supposed to work with default values and this one is for a bit different scenario.

Reply:
Now I see that check_boolean_value() is probably not needed at all, because for bool will always return ON or OFF, so check_boolean_value() will always return true.

What different scenario are you referring to?

Reply:
> What different scenario are you referring to?
I meant that saved_value_to_string() is supposed to set default value for a variable and here we set value provided by user.
As for this check_boolean_value()  - most probably it isn't needed now, I just wanted to minimize code changes. Do you think we should remove it at all?

Reply:
I think if it is not needed, we should remove it. What is your opinion @satya-bodapati ?

Reply:
Existing changes are anyway above minimal :) , OK to remove

Reply:
> I think if it is not needed, we should remove it.

Done. Still use it for default value case. It would require additional code changes to remove it.